modules/standard/Socket.chpl

```diff
@@ -0,0 +1,1282 @@
+                  any new connection
+*/
+proc tcpListener.accept(in timeout: timeval = new timeval(-1,0)):tcpConn throws {
+  var client_addr:sys_sockaddr_t = new sys_sockaddr_t();
+  var fdOut:fd_t;
+  var err_out:err_t = 0;
+  // try accept
+  err_out = sys_accept(socketFd, client_addr, fdOut);
+  // if error is not about blocking throw error
+  if err_out != 0 && err_out != EAGAIN && err_out != EWOULDBLOCK {
+    throw SystemError.fromSyserr(err_out, "accept() failed");
+  }
+  // successfully return file
+  if err_out == 0 {
+    return openfd(fdOut):tcpConn;
+  }
+  var localSync$: sync c_short;
+  // create event pending state
+  var internalEvent = event_new(event_loop_base, this.socketFd, EV_READ | EV_TIMEOUT, c_ptrTo(syncRWTCallback), c_ptrTo(localSync$):c_void_ptr);
+  defer {
+    // cleanup
+    event_free(internalEvent);
+  }
+  var t: Timer;
+  t.start();
+  // make event active
+  if timeout.tv_sec == -1 {
+    err_out = event_add(internalEvent, nil);
+  }
+  else {
+    err_out = event_add(internalEvent, c_ptrTo(timeout));
+  }
+  if err_out != 0 {
+    throw new Error("accept() failed");
+  }
+  // return value
+  var retval = localSync$.readFE();
+  // stop timer
+  t.stop();
+  // if error was timeout throw error
+  if retval & EV_TIMEOUT != 0 {
+    throw SystemError.fromSyserr(ETIMEDOUT, "accept() timed out");
+  }
+  var elapsedTime = t.elapsed(TimeUnits.microseconds):c_long;
+  // try accept again
+  err_out = sys_accept(socketFd, client_addr, fdOut);
+  if err_out != 0 {
+    // error was not about blocking wait so throw it
+    if err_out != EAGAIN && err_out != EWOULDBLOCK {
+      throw SystemError.fromSyserr(err_out, "accept() failed");
+    }
+    // no indefinitely blocking wait
+    if timeout.tv_sec == -1 {
+      var totalTimeout = timeout.tv_sec*1000000 + timeout.tv_usec;
+      // timer didn't elapsed
+      if totalTimeout > t.elapsed(TimeUnits.microseconds) {
+        const remainingMicroSeconds = ((totalTimeout - elapsedTime)%1000000);
+        const remainingSeconds = ((totalTimeout - elapsedTime)/1000000);
+        return this.accept(new timeval(remainingSeconds, remainingMicroSeconds));
+      }
+      throw SystemError.fromSyserr(err_out, "accept() failed");
+    }
+    return this.accept();
+  }
+  return openfd(fdOut):tcpConn;
+}
+
+/*
+  Close the file descriptor
+*/
+proc tcpListener.close() {
+  sys_close(this.socketFd);
+}
+
+/*
+  Returns the address on which socket is
+  listening on and bound to.
+
+  :return: bound address
+  :rtype: `ipAddr`
+*/
+proc tcpListener.addr throws {
+  return getsockname(this.socketFd);
+}
+
+inline operator !=(in lhs: tcpListener,in rhs: tcpListener) {
+  return lhs.socketFd != rhs.socketFd;
+}
+
+proc tcpListener.writeThis(f) throws {
+  f.write("(","addr:",this.addr,",fd:",this.socketFd);
+}
+
+pragma "no doc"
+extern const SOMAXCONN: int;
+/*
+  Default `backlog` value used in :proc:`listen`
+*/
+var BACKLOG_DEFAULT:uint(16);
```

Comment:
I would expect this to be `const` rather than `var`.

Also, see https://github.com/chapel-lang/chapel/blob/main/doc/rst/developer/bestPractices/StandardModuleStyle.rst#other-identifiers -- I would expect this variable to be named `backlogDefault` -- we don't use the underscore naming style in standard modules.